src/movies.js

```diff
@@ -1,22 +1,114 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data");
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+
+  const neW = movies.map((eachMovies)=>{
+    return eachMovies.director;
+  })
+
+  return neW
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies) {
+
+    const newArry = movies.filter((eachMovies)=>{
+        return eachMovies.genre.toLowerCase() === 'drama';
+    })
+
+    return eachMovies.genre;
+}
+
```

Comment:
- Aquí te está pidiendo que devuelvas **cuantas** películas de género Drama ha dirigido Steven Spielberg, por lo que tienes que tener en cuenta también la condición de que el director sea Steven Spielberg dentro del filter. Además, la propiedad `genre` es un array, por lo que tienes que comprobar si uno de los géneros de la película es drama utilizando el método de los arrays `includes()`:

- Aparte de eso, lo que devuelve la función tiene que ser un número que represente cuántas películas ha devuelto el `filter()`, por lo tanto tienes que hacer return de su `length`.

```suggestion
function howManyMovies(movies) {

  const newArry = movies.filter((eachMovie) => {
    return eachMovie.genre.includes('Drama') && eachMovie.director === 'Steven Spielberg';
  })

  return Number(newArry.length.toFixed(2));
}
```